### ISSUE TITLE
MoE distribution curve

### ISSUE THREAD

**Issue by gdcumming** [gfmoore/esci-precision#3]
This is the sampling distribution of MoE that would be obtained if a stated N is used by a researcher. This was added in ESCI intro, and wasn't in the earlier ESCI. Pp. 269-270 in ITNS explain, without calcs.

In UTNS the sampling distribution of s, the sample SD, which has the same shape (chi distribution NB not chi-sq) as the distribution of MoE, is discussed in two places. In Chap 10, pp 267-270, esp. Fig 10.3. That distribution is the smooth curve on the 's pile', which is the heap of s values from repeated sampling. (It's also in Chap 11, pp 299-300 but I that's a bit more complicated.) I'll send Chap 10. 

Fig 10.3 in UTNS uses the 'Sampling' sheet in ESCI chaps 10-13 that you have. You could play with running samples and seeing the s pile build. (Chap 11 uses the 'd heap' sheet.)

Neither book gives any formulas for the s curve or MoE curve, which are the same within a scale factor.

The basis is that 
(s^2 / sigma^2) * df is distributed as chi-squared(df)

Our target MoE axis is in units of f, so considering this as a variable, in units of obtained MoE.

For the indep groups case, df = (2N - 2) and MoE = t(.95)(df) * s * sqrt(2/N) because we assume the two groups are each of size N. See cell J90.

In ESCI intro, sheet 'Precision two', unprotect sheet and display headers and gridlines as before, click carefully on the heavy brown line to select it. See in the formula bar reference to $C$104:$C$1103 and similar in Column F.

Scroll down to see these two columns of values, which are the x and y values used to display that curve. Column C are the MoE values in 1000 small steps on the horiz axis. Select one of these. Then click one cell to the right (Col D) and see the formula for MoE^2 / (a constant in cell C96 (red) based on f in J90). The squaring is equiv to shifting from s values on our X axis to s^2 values, whose distribution we know.

Click a further cell to the right (Col E) to see the CHIDIST value (E96 holds df). That's an area under the curve. To translate this into an ordinate (a y value), one cell further right (Col F) takes the difference between two of those area values and divides this by twice the MoE increment between points, i.e. twice the value in C99. That gives us a height. Multiple this by a convenient scale factor (in E98) and we have one of the y values for our curve.

I hope that makes sense... please ask...!

**Comment by gfmoore** [comment]
Man, it's midnight here and I've just cracked the first moe distribution curve - well at least it looks ok!! Nightmare. laughing my head off. (Need some emojis on these comments.)

Well see what you think. Now I'll need to go through the other three possibilities to see what changes - hopefully nothing except getting appropriate N

Bed time.

**Comment by gdcumming** [comment]
0.1.0
Great to see four nice-looking curves, nicely shaded. I haven't dug into the formulas yet...

**Comment by gdcumming** [comment]
0.1.8

As I just noted, I still need to study this. But one observation and a suggestion.

For Paired, truncation at max, with assurance, there is a stripe across the bottom, the full width, with the curve not permitted to go fully down to the axis. In extreme cases, you can't eyeball that roughly 1% of area under curve is to the right of the cursor.

As a cross-check, can you find the 99th percentile of the area under the curve and compare that with the cursor position? Maybe the way it's calculated guarantees a match. But perhaps worth looking at?

**Comment by gfmoore** [comment]
Yeah it was a visual thing where the orange of the fill and brown of the curve was obliterating the axis. I'll probably have to redraw the axis.

⁣Regards,

Gordon Moore
gm@gordonmoore.co.uk​

**Comment by gdcumming** [comment]
0.1.9

I had a squiz this morning and I think you must have amended the way the curve is displayed, or maybe are still working on it. Looks good.

I'm not sure it ever made it to any spec, but it would be good in due course to have a vertical line a bit to the left of the curve, as a probability density axis, also the N for the curve shown on the curve itself. -- both more or less as in ESCI. No doubt you had these in mind.... sorry.

**Comment by gdcumming** [comment]
0.1.9

I've had a good play--great fun! I can't say I follow every detail of your calcs for the MoE curve, p. 4 in your "Precision for planning algorithms" doc. But what I follow does look good, and a close match to how ESCI does it.

It's hardly relevant, but I've also discovered that a new Excel function gives the pdf of the chi sq distribution directly, i.e. the ordinate values that we need for the plot of the curve:
CHISQ.DIST(x,df,FALSE) where 'false' says 'not cumulative', i.e gives the ordinate, whereas TRUE would give the area under.

Using that would simplify the Excel a bit.

There is also a formula for the pdf, which gives the ordinate in terms of x and df, and uses one exp() and one gamma() function.

I've also looked in 'ESCI intro' at where the cursor falls in relation to the curve, when 'with assurance' is selected. The cursor is indeed at the 99% point, i.e. at most 1% of area under the curve is to the right of the cursor. Note that, esp at smallish N, the fact that N needs to be an integer can mean that the right tail area is actually a good bit less than 1%. (I'd never thought to do that before, and it's a nice relief to discover that all's ok.)

I also compared the shape and approx spread of your curve and mine, for a range of rho and targ MoE values, including all the extremes. By eyeball they matched well.

Noting that height is arbitrary, I explored how heights looked at the full range of situations. Your pfp seems to me to good, sometimes the curve is a bit low, sometimes a bit high, but overall a pretty good compromise.

I guess it would still be good if you could do that check of the right tail area when assurance selected.

That, and the vertical line for prob density, and the N value on the curve (not turned off when Panel 4 clicked off), and I think we're done!

**Comment by gfmoore** [comment]
>> 'm not sure it ever made it to any spec, but it would be good in due course to have a vertical line a bit to the left of the curve, as a probability density axis, also the N for the curve shown on the curve itself. -- both more or less as in ESCI. No doubt you had these in mind.... sorry.

No, I just didn't do it, but have now.

**Comment by gfmoore** [comment]
>> As a cross-check, can you find the 99th percentile of the area under the curve and compare that with the cursor position? Maybe the way it's calculated guarantees a match. But perhaps worth looking at?

I have no idea how I will do that?

**Comment by gfmoore** [comment]
>>  I think you must have amended the way the curve is displayed

I did.

**Comment by gfmoore** [comment]
>> I guess it would still be good if you could do that check of the right tail area when assurance selected.

Re comment above. i.e. I don't know how to do that.

My code followed the Excel code, so if you can tell me how you do it for Excel I can probably do it for JavaScript, but not sure why it would help as my code follows what you did in Excel!!! (I just use different intervals for MoE)

**Comment by gfmoore** [comment]
>> It's hardly relevant, but I've also discovered that a new Excel function gives the pdf of the chi sq distribution directly, i.e. the ordinate values that we need for the plot of the curve:
CHISQ.DIST(x,df,FALSE) where 'false' says 'not cumulative', i.e gives the ordinate, whereas TRUE would give the area under.

>> Using that would simplify the Excel a bit.

I think that's what I did using jStat?

Rcum = 1 - jStat.chisquare.cdf(f2, df);

?

**Comment by gfmoore** [comment]
Or perhaps I didn't!

**Comment by gfmoore** [comment]
>> There is also a formula for the pdf, which gives the ordinate in terms of x and df, and uses one exp() and one gamma() function.

What's that then 😄

**Comment by gdcumming** [comment]
0.1.12

To check the achievement of gamma = .99, in ESCI intro 'Precision two' sheet I set, for example, targ MoE = 0.40. Saw N = 65, for assurance. Scrolled down, keeping an eye on Column C, looking for 0.40.

Row 370, x = 0.4005 and Col E says RH tail area = .0072 and that the transition through .01 occurs just a couple of rows above. Nice!

Given how incestuous, or in-bred, all our calculations are by now, this may not be worth chasing up further. But I was happy that it did work in my Excel. I also mentioned the lesson that, given that N takes discrete values, out to the right of the display (small N) the RH tail area is often a fair bit less than .01. Even so, the stated N is the minimum that gives MoE no more than targ MoE on no less than 99% of occasions.

For the pdf formula I searched.

https://en.wikipedia.org/wiki/Chi-square_distribution
section 1.2

https://wiki.geogebra.org/en/ChiSquared_Command

A nifty video showing how the formula and the ESCI chisq.dist agree (phew!):
https://www.youtube.com/watch?v=68t1b0KnXOo

We may be done here? Close?
